### PR TITLE
Add token auth marathon

### PR DESCRIPTION
**Pull request by jkorzekwa-atlassian**
*Note: Please remember to review the Datadog [Contribution Guidelines](https://github.com/DataDog/dd-agent/blob/master/CONTRIBUTING.md)
if you have not yet done so.*

### What does this PR do?

- Add tasksHealthy and tasksUnhealthy to metrics sent do datadog for all marathon apps
- Add ability to use ACS token to authenticate with the marathon api for marathon running in DC/OS platform 1.8

### Motivation

In order to alert on unhealthy marathon apps, we required the ability to compare unhealthy instances vs healthy instances for an app running in marathon, and alert based on a threshold. This required the tasksHealthy and tasksUnhealthy metrics provided by the marathon api

Marathon running in DC/OS 1.8 requires an ACS token for authentication. Added the ability to optionally configure the base url for the ACS auth endpoint. If configured, the user and password parameters will be used to generate an ACS auth token for the authentication header, and not used for basic auth.

### Testing Guidelines

An overview on [testing](https://github.com/DataDog/dd-agent/blob/master/tests/README.md)
is available in our contribution guidelines.

### Additional Notes

Anything else we should know when reviewing?

### CODE REVIEW

**Comment by irabinovitch** [comment]
CC @DanielLanger @hkaj

**Comment by jkorzekwa-atlassian** [comment]
@irabinovitch do you know when this might be merged and how long it will take to make it to the production dd-agent?

**Comment by hkaj** [comment]
Hi @jkorzekwa-atlassian 
Thank you for contributing!
Sorry for the delay here, I'll review and test it early next week. I gave it a first quick look and that's something we'll want to ship asap, so I'm adding it to the 5.11 milestone.
Thanks again

File: checks.d/marathon.py [hunk 1]
```diff
@@ -39,6 +41,7 @@ def check(self, instance):
         url = instance['url']
         user = instance.get('user')
         password = instance.get('password')
+        acs_url = instance.get('acs_url')
```
Comment (jkorzekwa-atlassian) [inline checks.d/marathon.py]: Added this to marathon.example.yaml

File: checks.d/marathon.py [hunk 2]
```diff
@@ -59,30 +62,52 @@ def check(self, instance):
...
+            'headers': {},
+            'auth': auth
+        }
+        if acs_url:
+            try:
+                auth_body = {
+                    'uid': auth[0],
+                    'password': auth[1]
+                }
+                r = requests.post(urljoin(acs_url, "acs/api/v1/auth/login"), json=auth_body, verify=False)
```
Comment (jkorzekwa-atlassian) [inline checks.d/marathon.py]: How would you suggest caching the token? I assume a new Marathon object is created on each check, which means I cant store the token in the class object itself.

Comment (jkorzekwa-atlassian) [inline checks.d/marathon.py]: Ok, added some logic to cache the acs token after it is grabbed for the first time, and only attempt to refresh when the token fails to auth.

**Comment by jkorzekwa-atlassian** [comment]
@hkaj Is there anything else you need from me to get this pushed through?

**Comment by hkaj** [comment]
Hey @jkorzekwa-atlassian 
Thanks for the update. The code looks good to me, I'll give it a spin in Azure and it should be good to go for 5.11. Will keep you updated here.

**Comment by hkaj** [comment]
Hi @jkorzekwa-atlassian i fixed the conflicts on another branch and am now trying to test this against a DC/OS cluster. Are you running an open source DC/OS cluster or the enterprise offer? I'm having trouble replicating a setup where this would make sense with the OS version.

**Pull request merged**